src/averageAges.js

```diff
@@ -55,12 +68,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const motherWithSon = people.filter(child =>
+    onlyWithSon
+      ? isMotherOfChild(people, child) && child.sex === 'm'
+      : isMotherOfChild(people, child)
+  );
+
+  return motherWithSon.reduce((sum, child) => sum
+  + child.born - getMotherBirthYear(people, child), 0) / motherWithSon.length;
+}
+
+function isMotherOfChild(people, child) {
```

Comment:
```suggestion
function getMotherOfChild(people, child) {
```
The name is a bit confusing as the function returns not a boolean